addons/web/static/tests/core/utils/numbers_tests.js

```diff
@@ -137,3 +141,144 @@ QUnit.module("utils", () => {
         assert.strictEqual(floatIsZero(-0.0000001, 7), false);
     });
 });
+
+QUnit.module("utils", (hooks) => {
+    hooks.beforeEach(() => {
+        patchWithCleanup(localization, { ...defaultLocalization, grouping: [3, 0] });
+    });
+
+    QUnit.module("numbers");
+
+    QUnit.test("formatFloat", function (assert) {
+        assert.strictEqual(formatFloat(false), "");
```

Comment:
This is probably field-specific, I don't think this result is desirable from the standalone util

Reply:
Should I add this line only for the field or drop it? 

Also, do you know if [those lines](https://github.com/odoo-dev/odoo/blob/master-web-frontend_improvement-vcr/addons/web/static/src/core/utils/numbers.js#L180-L182) only apply to fields? I feel like it's redundant with [this line](https://github.com/odoo-dev/odoo/blob/739974c418a3baea1beebc1589be1a7a0da9adc1/addons/web/static/src/core/utils/numbers.js#L195). If I remove the latter, I'll also need to change the[ call to `humanNumber`](https://github.com/odoo-dev/odoo/blob/739974c418a3baea1beebc1589be1a7a0da9adc1/addons/web/static/src/core/utils/numbers.js#L184).

I think the same remarks apply to `formatCurrency`, what about it?

Reply:
Yes, checking for "false" is field-specific, it corresponds to a null value in DB. This part of the test(s) should be moved to the field tests. I'm not sure what you mean with it being redudant with that line, do you mean to fallback to 0? I don't know if that one is useful, I guess it's for cases that pass in a null/undefined/empty string but that also seems like it shouldn't be supported by the util, and I'm not sure that it can happen in the context of a field, so maybe that fallback can be removed.